src/cleanfigure.m

```diff
@@ -322,6 +345,152 @@ function pruneOutsideBox(meta, handle)
+    if isYlog
+        yData = log10(yData);
+    end
+    isZlog = strcmp(get(meta.gca, 'ZScale'), 'log');
+    if isZlog && is3D
+        zData = log10(zData);
+    end
+
+    % If this is a 3D plot, project the data into the image plane
+    if is3D
+        % Get the projection angle
+        [az, el] = view(meta.gca);
+
+        % Projection matrix from view
+        % C        = viewmtx(az, el);
+        % NOTE: This is a subset of the MATLAB viewmtx function, as octave
+        % does not provide such functionality
+        % Make sure az and el are in the correct range.
+        el = rem(rem(el+180,360)+360,360)-180; % Make sure -180 <= el <= 180
+        if el>90,
+          el = 180-el;
+          az = az + 180;
+        elseif el<-90,
+          el = -180-el;
+          az = az + 180;
+        end
+        az = rem(rem(az,360)+360,360); % Make sure 0 <= az <= 360
+
+        % Convert from degrees to radians.
+        az = az*pi/180;
+        el = el*pi/180;
+
+        % View transformation matrix:
+        % Formed by composing two rotations:
+        %   1) Rotate about the z axis -AZ radians
+        %   2) Rotate about the x axis (EL-pi/2) radians
+
+        C = [ cos(az)           sin(az)           0       0
+             -sin(el)*sin(az)   sin(el)*cos(az)   cos(el) 0
+              cos(el)*sin(az)  -cos(el)*cos(az)   sin(el) 0
+              0                 0                 0       1 ];
+
+        % Put the data into the 4D datavector used in the projection
+        data     = [xData(:),...
+                    yData(:),...
+                    zData(:),...
+                    ones(size(xData(:)))];
+
+        % Project the data into the image plane
+        data     = C*data';
+
+        % Only take the x and y coordinates
+        xData    = data(1, :)';
+        yData    = data(2, :)';
+    else
+        % Turn the data into a row vector
+        xData    = xData(:);
+        yData    = yData(:);
+    end
+end
+% =========================================================================
+function [xLim, yLim] = getVisibleLimits(meta)
+    % Check whether this is a 3D plot
+    is3D = isAxis3D(meta.gca);
+
+    % Get the axis limits
+    xLim     = xlim(meta.gca);
+    yLim     = ylim(meta.gca);
+    if is3D
+        zLim = zlim(meta.gca);
+    end
+
+    % Check for logarithmic scales
+    isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+    if isXlog
+        xLim  = log10(xLim);
+    end
+    isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+	if isYlog
+        yLim  = log10(yLim);
+	end
+    isZlog = strcmp(get(meta.gca, 'ZScale'), 'log');
+    if isZlog && is3D
+        zLim  = log10(zLim);
+    end
+
+    % If this is a 3D plot, project the limits into the image plane
+    if is3D
+        % Get the projection angle
+        [az, el] = view(meta.gca);
+
+        % Projection matrix from view
+        % C        = viewmtx(az, el);
+        % NOTE: This is a subset of the MATLAB viewmtx function, as octave
```

Comment:
Yeah, i only took parts of the function. However, it will be some time until I can have a look on that due to travelling.

As far as i see this is just the product of the 3D rotational matrices around x and z axis
(See https://en.wikipedia.org/wiki/Rotation_matrix)